File: app/src/main/java/com/quranapp/android/frags/onboard/FragOnBoardThemes.java
/*
 * Copyright (c) Faisal Khan (https://github.com/faisalcodes)
 * Created on 31/3/2022.
 * All rights reserved.
 */

package com.quranapp.android.frags.onboard;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.quranapp.android.activities.ActivityOnboarding;
import com.quranapp.android.databinding.LytThemeExplorerBinding;
import com.quranapp.android.utils.app.ThemeUtils;
import com.quranapp.android.utils.sharedPrefs.SPAppConfigs;

import kotlin.Unit;

public class FragOnBoardThemes extends FragOnBoardBase {
    public static FragOnBoardThemes newInstance() {
        return new FragOnBoardThemes();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return LytThemeExplorerBinding.inflate(inflater, container, false).getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        LytThemeExplorerBinding mThemeBinding = LytThemeExplorerBinding.bind(view);
        setupThemeBinding(mThemeBinding);
    }

    private void setupThemeBinding(LytThemeExplorerBinding binding) {
        binding.themeGroup.setOnCheckChangedListener((button, checkedId) -> {
            if (getActivity() instanceof ActivityOnboarding) {
                ((ActivityOnboarding) getActivity()).mThemeChanged = true;
            }
            final String themeMode = ThemeUtils.resolveThemeModeFromId(checkedId);
            SPAppConfigs.setThemeMode(button.getContext(), themeMode);

            return Unit.INSTANCE;
        });
    }
}

File: app/src/main/java/com/quranapp/android/frags/onboard/FragOnboardThemes.kt
package com.quranapp.android.frags.onboard

import ThemeUtils
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatDelegate
import com.quranapp.android.R
import com.quranapp.android.databinding.LytThemeExplorerBinding
import com.quranapp.android.utils.sharedPrefs.SPAppConfigs.setThemeMode

class FragOnboardThemes : FragOnboardBase() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.lyt_theme_explorer, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        LytThemeExplorerBinding.bind(view).themeGroup.let {
            it.check(resolveThemeIdFromMode())
            it.onCheckChangedListener = { button, checkedId ->
                val themeMode = resolveThemeModeFromId(checkedId)
                setThemeMode(button.context, themeMode)
                AppCompatDelegate.setDefaultNightMode(ThemeUtils.resolveThemeModeForDelegate())
            }
        }
    }

    fun resolveThemeIdFromMode(): Int {
        return when (ThemeUtils.getThemeMode()) {
            ThemeUtils.THEME_MODE_DARK -> R.id.themeDark
            ThemeUtils.THEME_MODE_LIGHT -> R.id.themeLight
            ThemeUtils.THEME_MODE_DEFAULT -> R.id.systemDefault
            else -> R.id.systemDefault
        }
    }

    fun resolveThemeModeFromId(id: Int): String {
        return when (id) {
            R.id.themeDark -> {
                ThemeUtils.THEME_MODE_DARK
            }

            R.id.themeLight -> {
                ThemeUtils.THEME_MODE_LIGHT
            }

            else -> {
                ThemeUtils.THEME_MODE_DEFAULT
            }
        }
    }

}
